spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -274,18 +275,21 @@ public boolean canDeleteWhere(Filter[] filters) {
       }
     }
 
-    return deleteExpr == Expressions.alwaysTrue() || canDeleteUsingMetadata(deleteExpr);
+    return selectsPartitions(deleteExpr) || canDeleteUsingMetadata(deleteExpr);
```

Comment:
[minor] should we still keep `deleteExpr == Expressions.alwaysTrue()` it can even skip selectPartitions call

Reply:
I'd keep a simpler return statement in this case as the selects partition call should be fairly cheap for a true literal. Previously, we wanted to avoid planning in some obvious cases. The new check should be instant.